### PR TITLE
Code refactoring

### PR DESCRIPTION
**Pull request by hakob7475**

### CODE REVIEW

File: os-server/src/main/java/app/openschool/auth/AuthServiceImpl.java [hunk 1]
```diff
@@ -81,37 +74,31 @@ public UserDetails loadUserByUsername(String email) throws UsernameNotFoundExcep
...
   @Override
-  public void updateResetPasswordToken(String email) {
-    User user =
-        userRepository.findByEmail(email).orElseThrow(() -> new EmailNotExistsException(email));
-    if (resetPasswordTokenRepository.findByUser(user.getId()).isPresent()) {
-      ResetPasswordToken currentToken = resetPasswordTokenRepository.findByUser(user.getId()).get();
-      resetPasswordTokenRepository.delete(currentToken);
-    }
+  public void updateResetPasswordToken(String email, User user) {
+    Optional<ResetPasswordToken> currentToken =
```
Comment (SevakMart) [inline os-server/src/main/java/app/openschool/auth/AuthServiceImpl.java:78]: inline variable

File: os-server/src/main/java/app/openschool/auth/api/annotation/PasswordsMatch.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+import javax.validation.Constraint;
+import javax.validation.Payload;
+
+@Target(TYPE)
+@Retention(RUNTIME)
+@Constraint(validatedBy = PasswordMatchValidator.class)
+@Documented
+public @interface PasswordsMatch {
```
Comment (SevakMart) [inline os-server/src/main/java/app/openschool/auth/api/annotation/PasswordsMatch.java:17]: Let's make this a bit generic. Can be @SameAs and receive  the field name that should be match as parameter

File: os-server/src/main/java/app/openschool/course/api/dto/UserEnrolledCourseDto.java [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
 import java.time.LocalDate;
 
-public class UserCourseDto {
+public class UserEnrolledCourseDto {
```
Comment (SevakMart) [inline os-server/src/main/java/app/openschool/course/api/dto/UserEnrolledCourseDto.java:5]: why this is renamed?

**Review by SevakMart — CHANGES_REQUESTED** [review]
please fix comments

File: os-server/src/main/java/app/openschool/course/api/dto/UserEnrolledCourseDto.java (hunk 3 above)
Comment (hakob7475) [reply os-server/src/main/java/app/openschool/course/api/dto/UserEnrolledCourseDto.java:5]: To be different from dto of mentor's courses.

File: os-server/src/main/java/app/openschool/auth/api/annotation/processor/SameValuesValidator.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    fieldNames = Arrays.asList(sameValues.sameFields());
+  }
+
+  @Override
+  public boolean isValid(Object targetObject, ConstraintValidatorContext ctx) {
+    Set<Object> uniqueFieldValues =
+        fieldNames.stream()
+            .map(
+                fieldName -> {
+                  try {
```
Comment (SevakMart) [inline os-server/src/main/java/app/openschool/auth/api/annotation/processor/SameValuesValidator.java:27]: please move this lambda logic into a separate method, and create a new runtime exception for this case

File: os-server/src/main/java/app/openschool/common/event/SendResetPasswordEmailEvent.java [hunk 5]
```diff
@@ -0,0 +1,30 @@
+package app.openschool.common.event;
+
+import org.springframework.context.ApplicationEvent;
+
+public class SendResetPasswordEmailEvent extends ApplicationEvent {
+  String email;
```
Comment (SevakMart) [inline os-server/src/main/java/app/openschool/common/event/SendResetPasswordEmailEvent.java:6]: private

File: os-server/src/main/java/app/openschool/common/event/SendResetPasswordEmailEvent.java [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    super(source);
+    this.email = email;
+    this.token = token;
+  }
+
+  public String getEmail() {
+    return email;
+  }
+
+  public void setEmail(String email) {
```
Comment (SevakMart) [inline os-server/src/main/java/app/openschool/common/event/SendResetPasswordEmailEvent.java:19]: remove setters

**Pull request merged**